### PR TITLE
 add TypeScript language server

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
add TypeScript language server
### What issues does this PR fix or reference?
#3119

### CODE REVIEW

**Comment by tolusha** [comment]
A lot of JavaDoc missed.

**Comment by tolusha** [comment]
What about removing `js` prefix in names?
For instance: 
* `ls-typescript` instead of `ls-js-ts`
* `org.eclipse.che.ls.typescript.binaries` instead of `org.eclipse.che.ls.js-ts.binaries`

File: plugins/plugin-web/che-plugin-web-ext-shared/src/main/java/org/eclipse/che/plugin/web/shared/Constants.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.web.shared;
+
+/**
+ *
+ */
```
Comment (azatsarynnyy) [inline plugins/plugin-web/che-plugin-web-ext-shared/src/main/java/org/eclipse/che/plugin/web/shared/Constants.java:15]: empty javadoc

File: plugins/plugin-web/che-plugin-web-ext-shared/src/main/java/org/eclipse/che/plugin/web/shared/Constants.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public static final String LANGUAGE  = "language";
+
+    /** TS Project Type ID */
+    public static String TS_PROJECT_TYPE_ID = "typescript";
+
+    /** TS Language */
+    public static String TS_LANG = "typescript";
+
+    /** Default extension for TS files */
+    public static String TS_EXT = "ts";
```
Comment (azatsarynnyy) [inline plugins/plugin-web/che-plugin-web-ext-shared/src/main/java/org/eclipse/che/plugin/web/shared/Constants.java:27]: It seems that this constant is never used so may be remove it?

Comment (evidolob) [reply plugins/plugin-web/che-plugin-web-ext-shared/src/main/java/org/eclipse/che/plugin/web/shared/Constants.java:27]: Now it's used

**Review by vparfonov — APPROVED** [review]

**Pull request merged**